### PR TITLE
Adding troubleshooting docs items

### PR DESCRIPTION
**Pull request by chrisnegus**
**1. Issue, if available:**
Added troubleshooting items related to [Issue #1084](https://github.com/aws/karpenter/issues/1084), [Issue #1180](https://github.com/aws/karpenter/issues/1180) and [Issue #607](https://github.com/aws/karpenter/issues/607), as well as several slack discussions.

**2. Description of changes:**
The following troubleshooting entries were added:
* Daemonsets can result in deployment failures (@felix-zhe-huang, please review)
* Unspecified resource requests cause scheduling/bin-pack failures (@tzneal, please review)
* Missing discovery tags causing provisioning failures (@bwagner5, please review)

**4. Does this change impact docs?**
- [X] Yes, PR includes docs updates

### CODE REVIEW

File: website/content/en/preview/troubleshooting.md
Comment (tzneal) [inline website/content/en/preview/troubleshooting.md:107]:
```suggestion
Not using the Kubernetes [LimitRanges](https://kubernetes.io/docs/concepts/policy/limit-range/) feature to enforce minimum resource request sizes will allow pods with very low or non-existent resource requests to be scheduled.  This can cause issues as Karpenter bin-packs pods based on the resource requests.  If the resource requests do not reflect the actual resource usage of the pod, Karpenter will place too many of these pods onto the same node resulting in the pods getting CPU throttled or terminated due to the OOM killer.  This behavior is not unique to Karpenter and can also occur with the standard `kube-scheduler` with pods that don't have accurate resource requests.
```

Comment (tzneal) [reply website/content/en/preview/troubleshooting.md:107]: Do we have any best practice docs that recommend sizing of limits vs requests?

Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md:107]: @tzneal We have a general discussion on limits and pointers to Kubernetes docs on the subject in the [Karpenter Best Practices](https://aws.github.io/aws-eks-best-practices/karpenter/#use-limitranges-to-configure-defaults-for-resource-requests-and-limits) guide. Do you have suggestions for how to set particular limits?

Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md]: I incorporated your other comments.

Comment (tzneal) [reply website/content/en/preview/troubleshooting.md]: I don't know a good limit to request ratio value to recommend, probably worth punting on for now in that case as I can't find any recommendations from K8s or any existing EKS best practices either.

File: website/content/en/preview/troubleshooting.md
Comment (bwagner5) [inline website/content/en/preview/troubleshooting.md:128]: This reads like those specific tags are required, but any tags can be used. We just use the `karpenter.sh/discovery` tags in the getting started guide

File: website/content/en/preview/troubleshooting.md
Comment (bwagner5) [inline website/content/en/preview/troubleshooting.md:90]:
```suggestion
## DaemonSets can result in deployment failures
```

File: website/content/en/preview/troubleshooting.md
Comment (bwagner5) [inline website/content/en/preview/troubleshooting.md:92]:
```suggestion
For Karpenter versions 0.5.3 and earlier, daemonsets were not properly considered when provisioning nodes.
```

File: website/content/en/preview/troubleshooting.md
Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md:128]: Fixed.

File: website/content/en/preview/troubleshooting.md
Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md:92]: Good catch. Fixed.

File: website/content/en/preview/troubleshooting.md
Comment (bwagner5) [inline website/content/en/preview/troubleshooting.md:118]: `securityGroupSelector` is only required when using Karpenter's generated launch templates. If a user provides a custom launch template, then specifying a `securityGroupSelector` will actually cause a validation error. `subnetSelector` is always required though.  

Other than that, lgtm!

File: website/content/en/preview/troubleshooting.md
Comment (felix-zhe-huang) [inline website/content/en/preview/troubleshooting.md:98]: This doesn't sound right to me. The problem for Daemon set before v0.5.3 is that Karpenter doesn't consider the resource requests of daemon sets when selecting instance type. Karpenter will launch a smaller node and pod will fail to run due to not enough resources. 

The above error log happens when Karpenter bin-packing is filtering instance types according to just the Daemon-set resource requests. It is not an indicator of the problem.

File: website/content/en/preview/troubleshooting.md
Comment (felix-zhe-huang) [inline website/content/en/preview/troubleshooting.md:100]: I will suggest to emphasize the workaround is for before v0.5.6.

File: website/content/en/preview/troubleshooting.md
Comment (felix-zhe-huang) [inline website/content/en/preview/troubleshooting.md:103]: This is correct. It has been addressed. We will not see this problem anymore.

File: website/content/en/preview/troubleshooting.md
Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md]: Okay. I'll fix that. Thanks!

File: website/content/en/preview/troubleshooting.md
Comment (bwagner5) [inline website/content/en/preview/troubleshooting.md:123]: what do you think about adding aws CLI commands for users to check their subnet and security group selectors? 

```
$ aws ec2 describe-subnets --filters Name=tag:karpenter.sh/discovery,Values=${CLUSTER_NAME}
.. Returns subnets matching the selector ...

$ aws ec2 describe-security-groups --filters Name=tag:karpenter.sh/discovery,Values=${CLUSTER_NAME}
.. Returns security groups matching the selector ...
```

**Review by chrisnegus — COMMENTED** [review]
@felix-zhe-huang , @tzneal , @bwagner5 I've responded to all of the provided comments. Please let me know if further changes are required or provide /lgtm if everything looks okay.

File: website/content/en/preview/troubleshooting.md
Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md:123]: Sorry, I just saw this. Yes, I think it's a good idea and I'll add them.

Comment (chrisnegus) [reply website/content/en/preview/troubleshooting.md:134]: Done.

**Review by bwagner5 — APPROVED** [review]
lgtm! nicely written!

File: website/content/en/preview/troubleshooting.md
Comment (njtran) [reply website/content/en/preview/troubleshooting.md]: Looks like most instance types are [here](https://aws.amazon.com/ec2/instance-types/#General_Purpose) have a max  cpu:memory ratio of 1:4. GPUs have 1:8. If we need a ratio, 1:4 would be a good place to start thinking about it.

**Pull request merged**
